crates/sui-types/src/sui_system_state.rs

```diff
@@ -553,6 +581,104 @@ impl Default for SuiSystemState {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq, JsonSchema)]
+pub enum SuiSystemState {
+    V1(SuiSystemStateInnerV1),
+}
+
+impl SuiSystemState {
+    /// Always return the initial (genesis) version of the system state object.
+    /// Genesis always uses this version regardless of the current version.
+    pub fn into_genesis_version(self) -> SuiSystemStateInnerV1 {
+        match self {
+            SuiSystemState::V1(inner) => inner,
+        }
+    }
+
+    /// Always treat this as the latest version and return it. If it's not, return Err.
+    /// NOTE: Update the return type of this function whenever we upgrade the system state type.
+    pub fn to_latest_version(self) -> SuiResult<SuiSystemStateInnerV1> {
```

Comment:
is the plan to update the return types whenever a newer version is added? It looks error-prone